apis/cluster/services/flux/composition.yaml

```diff
@@ -0,0 +1,98 @@
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: xservicesflux.azure.platformref.upbound.io
+  labels:
+    type: gitops
+spec:
+  writeConnectionSecretsToNamespace: upbound-system
+  compositeTypeRef:
+    apiVersion: azure.platformref.upbound.io/v1alpha1
+    kind: XServices
+  patchSets:
+    - name: Common
+      patches:
+        - type: FromCompositeFieldPath
+          fromFieldPath: metadata.labels
+          toFieldPath: metadata.labels
+        - type: FromCompositeFieldPath
+          fromFieldPath: metadata.annotations
+          toFieldPath: metadata.annotations
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.providerConfigRef.name
+          toFieldPath: spec.providerConfigRef.name
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.deletionPolicy
+          toFieldPath: spec.deletionPolicy
+  resources:
+    - name: releaseFlux
+      base:
+        apiVersion: helm.crossplane.io/v1beta1
+        kind: Release
+        spec:
+          rollbackLimit: 3
+          forProvider:
+            namespace: flux-system
+            chart:
+              name: flux2
+              url: https://github.com/fluxcd-community/helm-charts/releases/download/flux2-2.9.2/flux2-2.9.2.tgz
+              version: 2.9.2
+            values:
+              imageAutomationController:
+                create: false
+              imageReflectionController:
+                create: false
+      patches:
+        - type: PatchSet
+          patchSetName: Common
+        - fromFieldPath: spec.operators.flux.version
+          toFieldPath: spec.forProvider.chart.version
+        - type: CombineFromComposite
+          combine:
+            variables:
+              - fromFieldPath: spec.operators.flux.version
+              - fromFieldPath: spec.operators.flux.version
+            strategy: string
+            string:
+              fmt: https://github.com/fluxcd-community/helm-charts/releases/download/flux2-%s/flux2-%s.tgz
+          toFieldPath: spec.forProvider.chart.url
+    - name: syncFlux
+      base:
+        apiVersion: helm.crossplane.io/v1beta1
+        kind: Release
+        spec:
+          rollbackLimit: 3
+          forProvider:
+            namespace: flux-system
+            chart:
+              name: flux2-sync
+              repository: https://fluxcd-community.github.io/helm-charts
+              version: "1.6.2"
```

Comment:
is this version intentionally not exposed?

Reply:
only a Kustomization and GitRepo creation is behind this helm-chart `flux2-sync` we can do the expose later when we do the split - because this part is reusable between all platform-refs